rust/bridge/jni/src/lib.rs

```diff
@@ -202,20 +202,16 @@ pub unsafe extern "C" fn Java_org_signal_client_internal_Native_PreKeyBundle_1Ne
         let signed_prekey = native_handle_cast::<PublicKey>(signed_prekey_handle)?;
         let signed_prekey_signature = env.convert_byte_array(signed_prekey_signature)?;
 
-        let prekey = native_handle_cast_optional::<PublicKey>(prekey_handle)?.map(|k| *k);
-
-        let prekey_id = if prekey_id < 0 {
-            None
-        } else {
-            Some(jint_to_u32(prekey_id)?)
+        let prekey = match native_handle_cast_optional::<PublicKey>(prekey_handle)?.map(|k| *k) {
+            None => None,
+            Some(key) => Some((jint_to_u32(prekey_id)?, key)),
```

Comment:
May be clearer as `map`?

Reply:
I wasn't able to find a way using `map` that wasn't significantly more complicated/harder to read, maybe missing some obvious approach though?

Reply:
Mm, the `?` for `jint_to_u32` does make it tricky, huh. Okay, never mind.